### PR TITLE
CSS Modules

### PR DESCRIPTION
**Pull request by farism**
### Changelog
- `core-css` now depends on `core-icons` and includes it
- `core-react` now depends on `core-css`
- give `core-icons` it's own webpack config and use webpack-dev-server instead of custom express server
- Use `webpack-blocks` on `core-react`
- share `core-react` webpack config with `styleguidist.config.js`
- create custom `resolveUrlLoader` block to deal with relative font urls
- general cleanup

Related Issue: https://github.com/procore/core/issues/219

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -10,7 +10,9 @@
     "build": "lerna run --parallel build",
     "build:docs": "lerna run --parallel build:docs",
     "dev": "lerna run --parallel dev",
+    "clean": "lerna clean",
     "precommit": "lerna run lint-staged",
+    "prepublish": "npm run build",
```
Comment (dannyphillips) [inline package.json]: should this be `lerna build` or something similar?

File: packages/core-css/package.json [hunk 2]
```diff
@@ -12,7 +12,7 @@
     "dev:docs": "catalog start",
     "lint-staged": "lint-staged",
     "lint:scss": "stylelint ./src/components/*.scss",
-    "prepublishOnly": "yarn build",
+    "prepublish": "npm run build",
```
Comment (dannyphillips) [inline packages/core-css/package.json]: why not use yarn?

File: packages/core-css/src/core.scss [hunk 3]
```diff
@@ -46,3 +46,6 @@
 
 // Trumps
 @import "./trumps/base";
+
+// CORE Icons
+@import "~@procore/core-icons";
```
Comment (dannyphillips) [inline packages/core-css/src/core.scss]: does this import relative or from npm?

Comment (dannyphillips) [reply packages/core-css/src/core.scss]: if npm, how do we ensure core-css always consumes the correct version from npm?

eg. 2.4.1 [core-css] -> 2.4.1 [core-icons from npm]

File: packages/core-icons/webpack.config.js [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+    publicPath: '',
+  }),
+
+  css(),
+
+  match(['*.eot', '*.svg', '*.ttf', '*.woff', '*.woff2'], [file()]),
+
+  extractText('icons.css'),
+
+  env('development', [
```
Comment (dannyphillips) [inline packages/core-icons/webpack.config.js]: this reads so nice :)

File: packages/core-react/docs/styleguide/VersionSelect.jsx [hunk 5]
```diff
@@ -1,73 +1,79 @@
...
+          value={this.state.currentVersion}
+          placeholder="Select an option"
+        />
       </div>
-    );
+    )
   }
 }
 
-export default VersionSelect;
```
Comment (dannyphillips) [inline packages/core-react/docs/styleguide/VersionSelect.jsx]: how did prettier not catch this? Do I need to run prettier locally? install globally?

File: packages/core-react/docs/styleguide/template.html [hunk 6]
```diff
@@ -27,21 +17,13 @@
...
-		  .core-column span:hover {
+		  [class^="core-column"] span:hover {
 				color: #FFF;
 	      background-color: #f47e42;
 			}
 		</style>
 	</head>
 	<body>
-		<script>
-			document.addEventListener("DOMContentLoaded", function (event) {
```
Comment (dannyphillips) [inline packages/core-react/docs/styleguide/template.html]: where did this go?

File: packages/core-react/styleguide.config.js [hunk 7]
```diff
@@ -1,38 +1,6 @@
...
-              test: /\.css$/,
-              use: ['style-loader', 'css-loader'],
-            },
-            context.match
-          ),
-        ],
-      },
-    })
-}
+const webpackConfig = require('./webpack.config.js')
```
Comment (dannyphillips) [inline packages/core-react/styleguide.config.js]: I love this change :)

File: packages/core-css/src/core.scss (hunk 3 above)
Comment (mboperator) [reply packages/core-css/src/core.scss]: Tilde (~) imports from the `node_modules` directory

File: packages/core-react/docs/styleguide/template.html (hunk 6 above)
Comment (farism) [reply packages/core-react/docs/styleguide/template.html]: Shouldn't be needed anymore.

File: packages/core-react/docs/styleguide/VersionSelect.jsx (hunk 5 above)
Comment (farism) [reply packages/core-react/docs/styleguide/VersionSelect.jsx]: lint-staged may not be matching on this file path correctly

**Pull request merged**